src/django_idom/types.py

```diff
@@ -51,3 +51,17 @@ class ViewComponentIframe:
     view: View | Callable
     args: Sequence
     kwargs: dict
+
+
+@dataclass
+class QueryOptions:
```

Comment:
If this is a `TypedDict` then users don't need to import this class and can instead declare options using dict literals.

Reply:
I suppose that having options are the first pos arg doesn't suffer from this issue since it is either a dict or callable.

Reply:
I think that using an object here still simplifies some things later down the line. For example, if we ever change parameter names, it's good to generate exceptions.

Reply:
Closing this, as I believe strictly typed objects have significant benefits over loosely typed dicts.